### ISSUE TITLE
[Calendar] `Disabled` days, months and years should be focusable

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2994]
### Provide a general summary of the issue here

For the disabled days, months and years we should use `aria-disabled` instead of `disabled` to keep them focusable.

Please take a look at the comment from Kasia: https://github.com/sl-design-system/components/issues/2568#issuecomment-3872674993

### 👤 Your name

SLDS Team

### 🧢 Your Product/Team

SLDS
